pkg/reconciler/pipeline/controller.go

```diff
@@ -860,6 +860,61 @@ func (r *pipelineReconciler) updateDesiredState(ctx context.Context, pl *dfv1.Pi
 	}
 }
 
+func (r *pipelineReconciler) vertexResumeHandler(ctx context.Context, pl *dfv1.Pipeline) error {
+	log := logging.FromContext(ctx)
+
+	newDesiredPhase := dfv1.VertexPhaseRunning
+	// Check if the pipeline's resume strategy is set to "slow"
+	// If so, we want to remove the `.spec.replicas` field during resume,
+	// which will let the vertex fall back to its minReplicas setting.
+	removeReplicas := pl.GetAnnotations()[dfv1.KeyResumeStrategy] == string(dfv1.ResumeStrategySlow)
+
+	// Get all existing vertices that belong to the pipeline
+	existingVertices, err := r.findExistingVertices(ctx, pl)
+	if err != nil {
+		return err
+	}
+
+	// Loop over each vertex and apply patches as needed
+	for _, vertex := range existingVertices {
+		// Determine whether the desired phase needs to be updated
+		currentPhase := vertex.Spec.Lifecycle.GetDesiredPhase()
+		needsPhasePatch := currentPhase != newDesiredPhase
+
+		// if phase condition is not true, skip patching this vertex
+		if !needsPhasePatch {
+			continue
+		}
+
+		var patchJson string
+
+		// Construct the patch JSON string based on what needs to change
+		if removeReplicas {
+			// Remove replicas and update the desired phase
+			patchJson = fmt.Sprintf(`{"spec":{"replicas":null,"lifecycle":{"desiredPhase":"%s"}}}`, newDesiredPhase)
```

Comment:
We set `"replicas": null` here: what sets it back in order for it to work to unpause and resume next time around in "fast" mode? Is it the vertex autoscaler that will adjust it? 

Context: I just did a test in which I integrated it with [Numaplane PR](https://github.com/numaproj/numaplane/pull/833). I first set it to "slow" through PipelineRollout and paused/resumed. Observed that `replicas` field was removed on vertex. Then I did a PPND test of changing pipeline topology. Observed that pipeline annotation correctly got set to "fast" during the resume stage and then after got set back to "slow" per the PipelineRollout spec. Observed that vertex `replicas` field never changed. Just want to confirm this is expected behavior, thanks.

Reply:
cc @whynowy @vigith

Reply:
It's not expected to see the replicas go back to origin after the operation of `resume slow -> pause -> resume fast`.

Reply:
> When resume slow happens, the current replica is expected to be min or 1

Unfortunately, I don't have the window up anymore, but I believe I saw `replicas: null`. Is that what you mean?

Reply:
Sounds like we should be good then. Am I correct to assume that the autoscaler is the one which will then increase that value?

Reply:
> Sounds like we should be good then. Am I correct to assume that the autoscaler is the one which will then increase that value?

Yes, you are.